webapp/src/scriptmanager.tsx

```diff
@@ -424,11 +435,25 @@ export class ScriptManagerDialog extends data.Component<ScriptManagerDialogProps
 
         const zipName = `makecode-${targetNickname}-project-download.zip`
 
-        pxt.BrowserUtils.browserDownloadDataUri(datauri, zipName);
 
         this.setState({
             download: null
         });
+
+        pxt.BrowserUtils.browserDownloadDataUri(datauri, zipName);
+        if (pxt.BrowserUtils.isInGame()) {
+            const downloadJsx = this.renderDownloadDialog(zipName);
+
+            setTimeout(async () => {
```

Comment:
Is the timeout waiting for the download to finish? 2 seconds may not be enough time for someone with a whole bunch of projects and very slow internet (though not sure how likely that is).

Reply:
Yes, it is. That's a good point. I don't really want to increase the time for it off the bat, though, because if it's too long and the notification shows up _way_ after downloading then it feels even more disjoint. 

This is not my favorite solution, but I included the timeout because I couldn't find an event that could tell us that the user stopped interacting with the filesystem/download unfortunately. No timeout was even worse because the modal would show up even before the filesystem modal would show up.